default/AI/ProductionAI.py

```diff
@@ -62,7 +63,14 @@ def find_best_designs_this_turn():
         ShipDesignAI.Cache.print_best_designs()
 
 
-def get_design_cost(cur_turn, design, pid):
+def get_design_cost(design, pid):  # TODO: Use new framework
+    """Find and return the design_cost of the specified design on the specified planet.
+
+    :param design:
+    :param pid: int: planet id
+    :return: cost of the design
+    """
+    cur_turn = fo.currentTurn()
```

Comment:
just below these lines (in the main file, not this diff) is the placeholder treatment for location invariance.  That is now exposed, as a bool shipDesign property "costTimeLocationInvariant", so it would be good to put in the real check at this time.